lab-devon/lib/parse-message.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+module.exports = exports = {};
+
+exports.nickCommand = function nickCommand(message, socket) {
+  socket.username = message.split(' ').slice(1).join(' ').trim();
+  socket.write(`I am now calling you ${socket.username}\n`);
+};
+
+exports.dmCommand = function dmCommand(message, sockets) {
+  let toUsername = message.split(' ')[1];
+  let content = message.split(' ').slice(2).join(' ').trim();
+  sockets.forEach( s => {
+    if(s.username === toUsername)
+      s.write(toUsername + ':' + content);
```

Comment:
Couple issues with Line 15:

1. Let's say I have two users, Jack and Jill, in the chatroom. If Jack sends a ``dm`` to Jill by typing in ``dm Jill Hi Jill``, the message displays to Jill as ``Jack: Hi Jack``. 

So at this point, ``toUsername`` is equal to Jack, i.e. the person who's receiving the message, NOT the person who's sending the message, as it should be. I'll let you figure this out yourself for a bit, but come to me if you get stuck. What should be replacing ``toUsername`` instead in order to display the sender's username? 

2. You'll have different uses cases where it's more practical to just use a ``+`` sign verses the ES6 interpolation with the ``${}``, but I think this is a better case for ES6. The above would be more easily written as:

```
`${toUsername}: content`

```

Once you starting using it more, you'll see that it's easier to than keeping track of quotes and plus signs.